core/src/main/scala/org/apache/spark/Partitioner.scala

```diff
@@ -137,6 +137,18 @@ private[spark] class PartitionIdPassthrough(override val numPartitions: Int) ext
   override def getPartition(key: Any): Int = key.asInstanceOf[Int]
 }
 
+/**
+ * A [[org.apache.spark.Partitioner]] that partitions all records use partition value map
+ */
+private[spark] class PartitionValueMapPartitioner(
+    valueMap: mutable.Map[Seq[Any], Int],
```

Comment:
Instead of using `Seq[Any]`, can we use `InternalRowComparableWrapper` here?

Reply:
I see. I guess it also makes sense to keep `KeyGroupedPartitioner` in `spark-catalyst` or `spark-sql`, since the concept of `KeyGroupedPartitioning` only applies to Spark SQL.

Reply:
Ah OK, let's potentially do it as a follow-up then